src/components/inputs/Selector.tsx

```diff
@@ -13,21 +13,46 @@ export enum SelectorType {
   STATUS_SELECTOR = 'statusSelector',
 }
 
+interface customOptions {
+  id: string
+  name: string
+  value?: string
+  //other custom types..
+}
+
 interface Prop {
   getSelectedValue: (value: unknown) => void
   startIcon: ReactNode
   value: unknown
   selectorType: SelectorType.STATUS_SELECTOR | SelectorType.ASSIGNEE_SELECTOR
   options: unknown[]
   buttonContent: ReactNode
+  placeholder?: string
+  customOptions?: customOptions
 }
 
-export default function Selector({ getSelectedValue, startIcon, value, selectorType, options, buttonContent }: Prop) {
-  const [isOpen, setIsOpen] = useState(false)
+export default function Selector({
+  getSelectedValue,
+  startIcon,
+  value,
+  selectorType,
+  options,
+  buttonContent,
+  placeholder = 'Change status...',
+  customOptions,
+}: Prop) {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)
+
+  const handleClick = (event: React.MouseEvent<HTMLDivElement>) => {
+    setAnchorEl(anchorEl ? null : event.currentTarget)
+  }
+
+  const open = Boolean(anchorEl)
+  const id = open ? 'autocomplete-popper' : undefined
```

Comment:
use empty string rather than `undefined` for `html` attribute values

Reply:
Changed in e58dca661e43a72f2a6a73a0ca916d88191512a5 commit